### PR TITLE
Implement MoviesRepository and add unit test

### PR DESCRIPTION
**Pull request by lo1d1**
- Created IMoviesRepository interface and MoviesRepository implementation
- Implemented methods: GetAllAsync, GetByIdAsync, AddAsync, UpdateAsync, DeleteAsync
- Added unit test for MoviesRepository using EF Core InMemory provider
- Configured DataContext with DbSet<Movie>

### CODE REVIEW

File: Domain/Interfaces/IMoviesRepository.cs [hunk 1]
```diff
@@ -0,0 +1,22 @@
+﻿using SneakPeek.Models;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Domain.Interfaces
+{
```
Comment (01vadim10) [inline Domain/Interfaces/IMoviesRepository.cs:9]: I suggest using "file-scoped namespaces"

**Comment by 01vadim10** [comment]
DRY (Don't Repeat Yourself)
In MoviesRepositoryTests.cs, you have some redundant using statements (System.Collections.Generic, System.Linq, System.Threading.Tasks are imported multiple times). Remove duplicates to follow DRY.

File: Domain/Interfaces/IMoviesRepository.cs (hunk 1 above)
Comment (01vadim10) [reply Domain/Interfaces/IMoviesRepository.cs:9]: Your files (e.g., Domain/Interfaces/IMoviesRepository.cs, Persistence/Repositories/MoviesRepository.cs) are using block-scoped namespaces (namespace Foo { ... }). To meet your requirement, you should use file-scoped namespaces:

namespace Domain.Interfaces;
// interface code

namespace Persistence.Repositories;
// class code

This will make the code cleaner and more modern

File: Domain/Models/Movie.cs [hunk 2]
```diff
@@ -4,6 +4,7 @@ namespace SneakPeek.Models;
 
 public class Movie
 {
+    public readonly int? Id;
```
Comment (01vadim10) [inline Domain/Models/Movie.cs:7]: You made Id readonly and nullable. In EF Core, Id is usually public int Id { get; set; } for tracking and updates. Using readonly may cause issues for updates and EF internal mechanisms. Consider reverting to auto-property unless you have a specific use case.

File: SneakPeak.Tests/Repositories/MoviesRepositoryTests.cs [hunk 3]
```diff
@@ -0,0 +1,49 @@
+﻿using Microsoft.EntityFrameworkCore;
```
Comment (01vadim10) [inline SneakPeak.Tests/Repositories/MoviesRepositoryTests.cs:1]: There are many duplicate using statements in MoviesRepositoryTests.cs. Clean them up.

**Comment by 01vadim10** [comment]
Based on the current changes in the pull request, here are the files and areas affected that are relevant for testing:

**Affected code:**
- `IMoviesRepository` interface: CRUD methods for Movie
- `MoviesRepository` implementation: actual logic for those CRUD methods
- Registration of repository in DI
- Minor update to `Movie` model (Id property)
- Minor fix in `DataContext`
- Added a test for `GetAllMoviesAsync` in `MoviesRepositoryTests`

**Your current test coverage (from the PR diff):**
- Only `GetAllMoviesAsync_ReturnsAllMovies` is tested.

---

## Suggested Additional Tests

### For `MoviesRepository`

#### 1. GetMovieByIdAsync
- **Test getting an existing movie by id**
- **Test getting a non-existent movie by id (should return null)**

#### 2. AddMovieAsync
- **Test adding a movie and ensuring it's persisted**
- **Test adding a movie with missing/invalid data (optional, if validation exists)**

#### 3. UpdateMovieAsync
- **Test updating an existing movie and verifying the update**
- **Test updating a non-existent movie (should not throw, or should throw as per your logic)**

#### 4. DeleteMovieAsync
- **Test deleting an existing movie and making sure it's removed**
- **Test deleting a non-existent movie (should not throw, or should throw as per your logic)**

#### 5. Edge Cases
- **Test repository methods with an empty database (GetAllMoviesAsync returns empty, etc.)**

---

## Example Test Names

- `GetMovieByIdAsync_ReturnsCorrectMovie_WhenMovieExists`
- `GetMovieByIdAsync_ReturnsNull_WhenMovieDoesNotExist`
- `AddMovieAsync_AddsMovieToDatabase`
- `UpdateMovieAsync_UpdatesMovieProperties`
- `UpdateMovieAsync_DoesNothing_WhenMovieDoesNotExist`
- `DeleteMovieAsync_RemovesMovieFromDatabase`
- `DeleteMovieAsync_DoesNothing_WhenMovieDoesNotExist`
- `GetAllMoviesAsync_ReturnsEmptyList_WhenNoMoviesPresent`

---

## Why these tests?
- They directly correspond to the contract of your interface and the code added/changed in the PR.
- They ensure the repository behaves correctly for all CRUD operations and edge cases.
- They help achieve high code coverage for only the new/changed code.

File: src/SneakPeek/Program.cs [hunk 4]
```diff
@@ -19,9 +20,15 @@
 
 builder.Services.AddControllers();
 builder.Services.AddDbContext<DataContext>(opt => opt.UseInMemoryDatabase("Movies"));
-builder.Services.AddDatabaseDeveloperPageExceptionFilter();
+
+builder.Services.AddScoped<IMoviesRepository, MoviesRepository>();
+
 var app = builder.Build();
 
+if (builder.Environment.IsDevelopment())
```
Comment (01vadim10) [inline src/SneakPeek/Program.cs:28]: This should be before line
> var app = builder.Build();
